packages/@ember/-internals/metal/lib/injected_property.ts

```diff
@@ -62,12 +81,12 @@ function inject(type: string, ...args: any[]): Decorator | DecoratorPropertyDesc
   let decorator = computed({
     get: getInjection,
 
-    set(this: any, keyName: string, value: any) {
+    set(this: object, keyName: string, value: unknown) {
       defineProperty(this, keyName, null, value);
     },
   });
 
-  if (calledAsDecorator) {
+  if (isElementDescriptor(args)) {
```

Comment:
Having to call this twice is annoying, but probably worth it for the increased typing benefits. This isn't a hot path.

Reply:
Yeah, doesn't work on 4.5.

Reply:
I refactored this.